### PR TITLE
If the user has < 65 cores and passes in an affinitized range for the 0th CPU group, if valid, honor it.

### PR DESCRIPTION
**Pull request by mrsharm**
## Summary

For cases where the user is running on a machine with < 65 cores and has passed in a ``GCHeapAffinitizeRanges`` for _just_ the 0th CPU group, apply the affinitized range rather than returning a ``CLR_E_GC_BAD_AFFINITY_CONFIG_FORMAT``.  And fail if:

- Any other CPU group is passed in if the machine has < 65 cores.
- If a user passes in a range that's > max processors available if the machine has < 65 cores and 0 is passed in as the CPU group.

NOTE: If both the Affinitized Range and the Affinitized Mask are passed, the Affinitized Mask is used if there are no CPU groups available on the machine. 

## Implementation Details

1. The check to see if we can bypass the Affinitized Ranges is simply to check if the user passed in the 0th CPU Group and if CPU Groups are not available.
2. If CPU Groups are not available, the affinitization will be based on the Affinitized Mask; to facilitate this cleanly, we are applying the CPU Group Affinitized Range on the Affinitized Mask so that down the road, it'll be successfully used.

### CODE REVIEW

**Review by cshung — APPROVED** [review]
Looks good to me, thanks for getting it fixed!

**Comment by Maoni0** [comment]
@janvorli would you like to take a look?

File: src/coreclr/vm/gcenv.os.cpp [hunk 1]
```diff
@@ -1171,12 +1171,25 @@ bool GCToOSInterface::ParseGCHeapAffinitizeRangesEntry(const char** config_strin
         return false;
     }
 
+    // If the user passes in 0 as the CPU group and they don't have > 64 cores, 
+    // honor the afinitized range passed in by bypassing the check.                                                                       
+    bool bypass_cpu_range_check_p = !CanEnableGCCPUGroups() && group_number == 0;
```
Comment (janvorli) [inline src/coreclr/vm/gcenv.os.cpp:1176]: A micro-nit - we don't use the `_p` suffix for booleans in the runtime code.

File: src/coreclr/vm/gcenv.os.cpp [hunk 2]
```diff
@@ -1171,12 +1171,25 @@ bool GCToOSInterface::ParseGCHeapAffinitizeRangesEntry(const char** config_strin
         return false;
     }
 
+    // If the user passes in 0 as the CPU group and they don't have > 64 cores, 
+    // honor the afinitized range passed in by bypassing the check.
```
Comment (janvorli) [inline src/coreclr/vm/gcenv.os.cpp:1175]: A nit:
```suggestion
    // honor the affinitized range passed in by bypassing the check.                                                                       
```

**Review by janvorli — APPROVED** [review]
LGTM modulo the nits.

**Comment by mrsharm** [comment]
Thanks for the feedback, @janvorli! I have incorporated them all and will merge these changes once the CI completes.

**Pull request merged**